code/game/objects/items/weapons/RPD.dm

```diff
@@ -518,50 +520,62 @@ var/global/list/RPD_recipes=list(
 		show_menu(usr)
 
 
-/obj/item/weapon/pipe_dispenser/afterattack(atom/A, mob/user)
+/obj/item/weapon/pipe_dispenser/attack_obj(obj/A, mob/user)
 	if(!in_range(A,user) || loc != user)
-		return 0
+		return
 
 	if(!user.IsAdvancedToolUser())
 		user << "<span class='warning'>You don't have the dexterity to do this!</span>"
-		return 0
-
-	if(istype(A,/area/shuttle)||istype(A,/turf/open/space/transit))
-		return 0
-
-	//So that changing the menu settings doesn't affect the pipes already being built.
-	var/queued_p_type = p_type
-	var/queued_p_dir = p_dir
-	var/queued_p_flipped = p_flipped
+		return
 
 	switch(p_class)
 		if(PAINT_MODE) // Paint pipes
 			if(!istype(A,/obj/machinery/atmospherics/pipe))
 				// Avoid spewing errors about invalid mode -2 when clicking on stuff that aren't pipes.
 				user << "<span class='warning'>\The [src]'s error light flickers!  Perhaps you need to only use it on pipes and pipe meters?</span>"
-				return 0
+				return
 			var/obj/machinery/atmospherics/pipe/P = A
 			playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
 			P.add_atom_colour(paint_colors[paint_color], FIXED_COLOUR_PRIORITY)
 			P.pipe_color = paint_colors[paint_color]
 			user.visible_message("<span class='notice'>[user] paints \the [P] [paint_color].</span>","<span class='notice'>You paint \the [P] [paint_color].</span>")
 			//P.update_icon()
 			P.update_node_icon()
-			return 1
+			return
 		if(EATING_MODE) // Eating pipes
-			// Must click on an actual pipe or meter.
+			// Must click on an actual pipe sor meter.
 			if(istype(A,/obj/item/pipe) || istype(A,/obj/item/pipe_meter) || istype(A,/obj/structure/disposalconstruct))
 				user << "<span class='notice'>You start destroying pipe...</span>"
 				playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
 				if(do_after(user, 2, target = A))
 					activate()
 					qdel(A)
-					return 1
-				return 0
+					return
+				return
 
 			// Avoid spewing errors about invalid mode -1 when clicking on stuff that aren't pipes.
 			user << "<span class='warning'>The [src]'s error light flickers!  Perhaps you need to only use it on pipes and pipe meters?</span>"
-			return 0
+			return
+		else
+			return ..()
+
+/obj/item/weapon/pipe_dispenser/attack(atom/A, mob/user)
```

Comment:
I'm not sure what you're doing but I think you've fucked up the click chain really bad.